aptos-move/framework/aptos-framework/sources/staking_contract.spec.move

```diff
@@ -329,4 +359,104 @@ spec aptos_framework::staking_contract {
+        aborts_if update_voting_power_increase && validator_set.total_voting_power > 0
+                && validator_set.total_joining_power + amount > validator_set.total_voting_power * voting_power_increase_limit / 100;
+        let stake_pool = global<stake::StakePool>(pool_address);
+        let post post_stake_pool = global<stake::StakePool>(pool_address);
+        let value_pending_active = stake_pool.pending_active.value;
+        let value_active = stake_pool.active.value;
+        ensures amount != 0 && stake::spec_is_current_epoch_validator(pool_address) ==> post_stake_pool.pending_active.value == value_pending_active + amount;
+        ensures amount != 0 && !stake::spec_is_current_epoch_validator(pool_address) ==> post_stake_pool.active.value == value_active + amount;
+        let maximum_stake = config.maximum_stake;
+        let value_pending_inactive = stake_pool.pending_inactive.value;
+        let next_epoch_voting_power = value_pending_active + value_active + value_pending_inactive;
+        let voting_power = next_epoch_voting_power + amount;
+        aborts_if amount != 0 && voting_power > MAX_U64;
+        aborts_if amount != 0 && voting_power > maximum_stake;
+    }
+
+    spec schema PreconditionsInCreateContract {
+        requires exists<stake::ValidatorPerformance>(@aptos_framework);
+        requires exists<stake::ValidatorSet>(@aptos_framework);
+        requires exists<staking_config::StakingRewardsConfig>(@aptos_framework) || !std::features::spec_periodical_reward_rate_decrease_enabled();
+        requires exists<stake::ValidatorFees>(@aptos_framework);
+        requires exists<aptos_framework::timestamp::CurrentTimeMicroseconds>(@aptos_framework);
+        requires exists<stake::AptosCoinCapabilities>(@aptos_framework);
+    }
+
+    spec schema CreateStakePoolAbortsIf {
+        resource_addr: address;
+        operator: address;
+        voter: address;
+        contract_creation_seed: vector<u8>;
+
+        // verify account::create_resource_account()
+        let acc = global<account::Account>(resource_addr);
+        aborts_if exists<account::Account>(resource_addr) && (len(acc.signer_capability_offer.for.vec) != 0 || acc.sequence_number != 0);
+        aborts_if !exists<account::Account>(resource_addr) && len(bcs::to_bytes(resource_addr)) != 32;
+        aborts_if len(account::ZERO_AUTH_KEY) != 32;
+
+        // verify stake::initialize_stake_owner()
+        aborts_if exists<stake::ValidatorConfig>(resource_addr);
+        let allowed = global<stake::AllowedValidators>(@aptos_framework);
+        aborts_if exists<stake::AllowedValidators>(@aptos_framework) && !contains(allowed.accounts, resource_addr);
+        aborts_if exists<stake::StakePool>(resource_addr);
+        aborts_if exists<stake::OwnerCapability>(resource_addr);
+        aborts_if exists<account::Account>(resource_addr) && acc.guid_creation_num + 12 >= account::MAX_GUID_CREATION_NUM;
```

Comment:
Could you explain why 12 is used here?

Reply:
Because the create_stake_pool function calls the stake::initialize_stake_owner function, the stake::initialize_stake_owner function calls the stake::initialize_owner function, and 12 events' guids are created in the initialize_owner function.

Reply:
Could you add the comment to the spec to explain it?

Reply:
Yes, i could.